dotnet/src/VectorDataIntegrationTests/VectorDataIntegrationTests/CRUD/NoDataConformanceTests.cs

```diff
@@ -0,0 +1,184 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.Extensions.VectorData;
+using VectorDataSpecificationTests.Support;
+using VectorDataSpecificationTests.Xunit;
+using Xunit;
+
+namespace VectorDataSpecificationTests.CRUD;
+
+/// <summary>
+/// Tests CRUD operations using a model without data fields.
+/// </summary>
+public class NoDataConformanceTests<TKey>(NoDataConformanceTests<TKey>.Fixture fixture) where TKey : notnull
+{
+    [ConditionalFact]
+    public virtual Task GetAsyncReturnsInsertedRecord_WithVectors()
+        => this.GetAsyncReturnsInsertedRecord(includeVectors: true);
+
+    [ConditionalFact]
+    public virtual Task GetAsyncReturnsInsertedRecord_WithoutVectors()
+        => this.GetAsyncReturnsInsertedRecord(includeVectors: false);
```

Comment:
FWIW @roji has fixed https://github.com/microsoft/semantic-kernel/issues/10736 and we can use `[ConditionalTheory]` now:

```suggestion
    [ConditionalTheory]
    [InlineData(true)]
    [InlineData(false)]
    public virtual Task GetAsyncReturnsInsertedRecord(bool includeVectors)
        => this.GetAsyncReturnsInsertedRecord(includeVectors);
```

However, such approach makes it a little bit harder to change the behavior for specific value, so I absolutely don't insist on taking my suggestion.

Reply:
If we are going to change this pattern, let's do that in a dedicated PR for that.  Better not to mix too many things in one PR.